client/components/embed-container/index.jsx

```diff
@@ -112,6 +131,56 @@ function embedTumblr( domNode ) {
 	}, 30 );
 }
 
+function triggerJQueryLoadEvent() {
+	// force JetpackSlideshow to initialize, in case navigation hasn't caused ready event on document
+	window.jQuery( 'body' ).trigger( 'post-load' );
+}
+
+function createSlideshow() {
+	if ( window.JetpackSlideshow ) {
+		triggerJQueryLoadEvent();
+	}
+
+	loadAndRun( SLIDESHOW_URLS.JS, () => {
+		triggerJQueryLoadEvent();
+	} );
+}
+
+function embedSlideshow( domNode ) {
+	debug( 'processing slideshow for', domNode );
+
+	// set global variable required by JetpackSlideshow
+	window.jetpackSlideshowSettings = {
+		spinner: SLIDESHOW_URLS.SPINNER
+	};
+
+	// load CSS if it isn't alread there
+	if ( ! document.head.querySelector( `link[href="${ SLIDESHOW_URLS.CSS }"]` ) ) {
```

Comment:
could we handle this using a module level flag instead of querying the DOM every time?